xbmc/filesystem/posix/PosixFile.cpp

```diff
@@ -313,6 +314,18 @@ bool CPosixFile::Rename(const CURL& url, const CURL& urlnew)
   
   if (rename(name.c_str(), newName.c_str()) == 0)
     return true;
+
+  // rename across mount points - need to copy/delete
+  if (errno == EXDEV)
+  {
+    if (XFILE::CFile::Copy(name, newName))
+    {
+      if (XFILE::CFile::Delete(name))
```

Comment:
What benefit will we have if we delete copy and fail instead of leaving original in place and put warning to the log?
Many scripts/function will just fail instead of continue working. This could help if someone will try to move file from RO filesystem.
Anyway, it should be really rare situation. RO filesystem is good example.